plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -191,7 +191,7 @@
     public BackgroundHiveSplitLoader(
             Table table,
             AcidTransaction transaction,
-            Iterable<HivePartitionMetadata> partitions,
+            Iterator<HivePartitionMetadata> partitions,
```

Comment:
That's not an innocent change. `Iterable` is repeatable use. `Iterator` is single use only.
Please revert.

Reply:
It is only used to create a `ConcurrentLazyQueue` that is read only once. It felt like it is better to express "read once" semantics in the interface.

Reply:
This is fine I think. But the change Iterable->Iterator should be a separate preparatory commit.